src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  static VALUE_OF_BITES = 50;
+
+  bite(beast) {
+    if (beast instanceof Carnivore || beast.hidden) {
+      return;
+    } else {
+      beast.health -= Carnivore.VALUE_OF_BITES;
+    }
+
+    if (beast.health <= 0) {
+      const indexBeast = Animal.alive.findIndex((item) => item === beast);
+
+      Animal.alive.splice(indexBeast, 1);
```

Comment:
This implementation uses `findIndex` and `splice` for removing dead animals which is not optimal in terms of performance, especially if `Animal.alive` array gets large. Consider using `filter` to create a new array excluding the dead animal.